src/charm.py

```diff
@@ -174,17 +118,17 @@ def _pebble_layer(self) -> Dict:
                     "user": "gh_exporter",
                     "command": "/srv/gh_exporter/github-actions-exporter",
                     "environment": {
-                        "GITHUB_WEBHOOK_TOKEN": f"{self.model.config['github_webhook_token']}",
-                        "GITHUB_API_TOKEN": f"{self.model.config['github_api_token']}",
-                        "GITHUB_ORG": f"{self.model.config['github_org']}",
+                        "GITHUB_WEBHOOK_TOKEN": f"{self.state.github_webhook_token}",
+                        "GITHUB_API_TOKEN": f"{self.state.github_api_token}",
+                        "GITHUB_ORG": f"{self.state.github_org}",
                     },
                 }
             },
             "checks": {
```

Comment:
@gregory-schiano this is an interesting case, do you think this kind configuration should also be delegated to the workload module? At least the value of the `tcp` key?

Reply:
Well indeed the fact that the check is a TCP check is definitely something specific to the workload and that should be owned by it.
So we should perhaps model that by using the Pebble Check Model class see https://github.com/canonical/operator/blob/main/ops/pebble.py#L897

Reply:
Since this PR already has 3 approvals, I'll merge it and submit a new one with this change so it will be easier to review.